### PR TITLE
Docs - Add Permission State Provider section to Authorization.md

### LINKED ISSUE #8643: Document update for permission state provider system
Based on https://github.com/abpframework/abp/pull/8543

You can wait for https://github.com/volosoft/volo/issues/6386 to see how it is used.

### PR DESCRIPTION
**Pull request by enisn**
- Closes #8643

### CODE REVIEW

File: docs/en/Authorization.md [hunk 1]
```diff
@@ -236,6 +236,25 @@ When you write this code inside your permission definition provider, it finds th
...
+### Permission depending on a Condition
+
+Permission states can be managed according to a condition such as a Feature, Global Feature, or a custom condition. Any condition which implements `IPermissionStateProvider` can be added via calling `AddStateProvider()` method for **PermissionDefinition**.
+
+See an example below:
+
+```csharp
+myGroup.AddPermission("Author_Management")
+    // To depend on Global Feature
+    .AddStateProviders(new RequireGlobalFeaturesPermissionStateProvider("AuthorManagementFeature"));
```
Comment (hikalkan) [inline docs/en/Authorization.md]: There are shortcut extension methods for features and global features. example:

````csharp
.RequireGlobalFeatures("AuthorManagementFeature")
````

Also, you've used `;` at end of the lines, but continued with `.` in the next lines. That code doesn't compile. You can separate examples to keep it simple

File: docs/en/Authorization.md [hunk 2]
```diff
@@ -236,6 +236,25 @@ When you write this code inside your permission definition provider, it finds th
 
 > Tip: It is better to check the value returned by the `GetPermissionOrNull` method since it may return null if the given permission was not defined.
 
+### Permission depending on a Condition
+
+Permission states can be managed according to a condition such as a Feature, Global Feature, or a custom condition. Any condition which implements `IPermissionStateProvider` can be added via calling `AddStateProvider()` method for **PermissionDefinition**.
```
Comment (hikalkan) [inline docs/en/Authorization.md]: It is not clear what is "Permission states". We should explain it like, "You may want to disable a permission based on a condition. Disabled permissions are not visible on the UI and always returns `prohibited` when you check them...". We should write that in the beginning to make the reader to understand what they will achieve when they learn this feature.

File: docs/en/Authorization.md [hunk 3]
```diff
@@ -236,6 +236,25 @@ When you write this code inside your permission definition provider, it finds th
...
+```csharp
+myGroup.AddPermission("Author_Management")
+    // To depend on Global Feature
+    .AddStateProviders(new RequireGlobalFeaturesPermissionStateProvider("AuthorManagementFeature"));
+	// To depend on Feature
+    .AddStateProviders(new RequireFeaturesPermissionStateProvider("BookStore.AuthorManagement"));
+	// To depend on your custom implementation
+    .AddStateProviders(new MyCustomPermissionStateProvider());
+	
+```
```
Comment (hikalkan) [inline docs/en/Authorization.md]: There is also global state providers ([see](https://github.com/abpframework/abp/blob/dev/framework/src/Volo.Abp.Authorization.Abstractions/Volo/Abp/Authorization/Permissions/AbpPermissionOptions.cs#L11)) which is not explained in this document.

**Review by hikalkan — CHANGES_REQUESTED** [review]

File: docs/en/Authorization.md (hunk 3 above)
Comment (enisn) [reply docs/en/Authorization.md]: It's going to be long documentation, should we extract State Providers as a new document? @hikalkan ,

Or we can just place a part about how to implement & configure custom conditions end of the section

Comment (hikalkan) [reply docs/en/Authorization.md]: Being long is not a problem. I will review your changes, thanks.

**Comment by enisn** [comment]
It's ready to review now @hikalkan

**Comment by enisn** [comment]
Changed milestone and base, so this pr is closed.

See #8746

**Review by hikalkan — APPROVED** [review]

**Pull request merged**